Bio/Entrez/__init__.py

```diff
@@ -563,6 +565,9 @@ def _construct_params(params):
 In case of excessive usage of the E-utilities, NCBI will attempt to contact
 a user at the email address provided before blocking access to the
 E-utilities.""", UserWarning)
+    # Tell Entrez if we want to set a custom cache location.
+    if "cache" not in params:
+        params["cache"] = cache
```

Comment:
I don't think these lines are needed. IIRC, this is building a record of all the parameters to be encoding into the URL or post sent to the NCBI. They don't need to know about where we might be caching DTD files on the user's machine (and revealing this could leak information like their computer account username).

Reply:
@peterjc Ah I see. Will delete. I mistakenly thought they were for internal parameters, and it makes sense that they're for NCBI given tool & email.